packages/core/docs/styleComponents.md

```diff
@@ -0,0 +1,169 @@
+  `static get styles()` configuration property on the constructor of a webcomponent.
+  - **flexible markup**: By making the HTML structure purposely 'loose', multiple Design Systems can
+    be made compatible with the style component. When the default html/css structure is not flexible
+    enough, it can be extended by creating new BEM element selectors.
+
+## Why BEM?
+For css class naming, we use [BEM](http://getbem.com/) naming conventions. This helps you identify
+semantics of your markup at a quick glance, resulting in more readable and maintainable code.
+BEM provides us the following advantages:
+- **Clear namings** that convey meaning about semantics and structure at a quick glance
+- **Prevent collisions** within shadow roots. Larger shadow roots using multiple style components
+would easily collide when css selectors would be scoped to the shadow dom in which they were
+created. For instance, calling the header of a card '.header' instead of '.c-card__header'
+would make the card component not reusable in different contexts.
+- **Maximum flexibility** we can write css regardless of context. When we create a style component
+for a card, we can offer it as a style component only and later decide to offer it as a webcomponent.
+Or offer the style component
+as a more lower level api for advanced use cases and the webcomponent (that has limiting markup and
+styling options) for the majority of cases.
+Also, when we already have a webcomponent for the card and people want to create a custom card
+component, it might make more sense to write a new webcomponent instead of extending the existing
+card webcomponent.
+Having style components as described above, allows for maximum flexibility in these scenarios.
+Future CSS updates can then be done from a central place, without having to rewrite forked webcomponents.
+- **Performance** we aim for applications having a limited number of shadow roots and we consider
+shadow root creation a performance concern. Style components reduce the need for shadow roots.
+<!-- TODO: we definitely suffered here in the Polymer era. We might want to create some new perf
+tests for our lit-element based apps to verify this claim again. -->
+
+
+## Challenges with BEM in shadow dom
+Mapping BEM components to shadow root is not really straightforward:
+ 1. host styles should be rewritten to `:host {}` instead of `.my-block {}`
+ 2. slot styles be rewritten to `::slotted(.my-block__element)` instead of `.my-block__element`
+ 3. we should not [self-apply classes](https://developers.google.com/web/fundamentals/web-components/best-practices)
+ on host level. Apart from this being a bad practice, it will trigger linting errors in our setup.
+
+### Mapping host and slot styles
+The [CSS Module](https://github.com/w3c/webcomponents/issues/759) and [CSS Selector](https://github.com/w3c/csswg-drafts/issues/3714) proposals would allow us to reuse BEM components more easily within webcomponents.
```

Comment:
```suggestion
The [CSS Module](https://github.com/w3c/webcomponents/issues/759) and [CSS Selector](https://github.com/w3c/csswg-drafts/issues/3714) proposals would allow us to reuse BEM components more easily within web components.
```